File: code/f4251.pas
var a,b,c,p,q:integer;
begin
    readln(a,b,c);
    while not eof do begin
        readln(p,q);
        writeln(abs(a*p+b*q+c)/sqrt(a*a+b*b):0:1);
    end;
end.

File: code/f4251.py
import sys
from math import sqrt
a,b,c=[int(x) for x in input().split()]
for s in sys.stdin:
    p,q=[int(x) for x in s.split()]
    kq=abs(a*p+b*q+c)/sqrt(a*a+b*b)
    print("%.1f" % kq)